typing_extensions/src_py3/test_typing_extensions.py

```diff
@@ -0,0 +1,636 @@
+import sys
+import os
+import abc
+import contextlib
+import collections
+from unittest import TestCase, main, skipUnless
+from typing import TypeVar, Optional
+from typing import T, KT, VT  # Not in __all__.
+from typing import Tuple, List
+from typing import Generic
+from typing import get_type_hints
+from typing import no_type_check
+from typing_extensions import NoReturn, ClassVar, Type, NewType
+import typing
+import typing_extensions
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    import collections as collections_abc  # Fallback for PY3.2.
+
+TYPING_V2 = sys.version_info >= (3, 5, 1)
+TYPING_V3 = sys.version_info >= (3, 5, 3)
+TYPING_V4 = sys.version_info >= (3, 6, 1)
```

Comment:
We will have ``TYPING_V5`` very soon (but I think it is OK to not include it in this PR, just don't forget to add it later).